[Sun Zhiyuan Felix] iP

potatodudedude opened the pull request:
# Hello.
[***Dododo. do. dododo.***](https://en.wikipedia.org/wiki/Dodo) <ins>Huh?</ins> 👍 
> dododo
> -dodo
- Dodo 
- Dodo
- Dodo
1. Dodo
2. Dodo
3. Dodo

Check this out! `dodo`

```ruby
if (dodo) {
    dodo.do
}
```

Features:
- [x] ~Tasks~
- [ ]  ???

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

potatodudedude committed:
Removed Duke
Added Dodo

potatodudedude committed:
Added copycat and bye function

potatodudedude committed:
Added new class for tasks
Added list function

potatodudedude committed:
Changed how to read user input to accomodate more commands
Added methods in Task class for changing isDone boolean

potatodudedude committed:
Implemented marking and unmarking tasks

potatodudedude committed:
Added subclasses Todo, Event, Deadline of Task

potatodudedude committed:
Implemented todo, event and deadline commands

potatodudedude committed:
Added in automated testing for:
All basic functions
Catching some edge cases

potatodudedude committed:
Added exceptions to handle array out of bounds after splitting command strings
Added help function
Added exceptions for invalid commands

potatodudedude committed:
added delete command

potatodudedude committed:
Add data folder and storage.txt and add reading storage.txt on startup

juneha1120 commented on src/main/java/Dodo.java:
```diff
@@ -10,6 +10,14 @@ public static void markCommandCheck(String[] commands) throws DodoException {
 
+    public static void deleteCommandCheck(String[] commands) throws DodoException {
+        if (commands.length != 2)  {
+            throw new DodoException("Delete commands needs to be followed by single task number.\n" +
```
I like this simple and clear exception message! But shouldn't the line breaks be before the operators?

juneha1120 commented on src/main/java/Dodo.java:
```diff
@@ -10,6 +10,14 @@ public static void markCommandCheck(String[] commands) throws DodoException {
         }
     }
 
+    public static void deleteCommandCheck(String[] commands) throws DodoException {
```
I like this clear method naming! With some javadoc comments think this will look better to me.

juneha1120 commented on src/main/java/Dodo.java:
```diff
@@ -160,6 +171,25 @@ public static void main(String[] args) {
+                    Task target = tasks.get(targetNo);
+                    tasks.remove(targetNo);
+                    System.out.println("The following task has been destroyed:\n" + target.toString() +
+                            "\nYou now have " + tasks.size() + " task(s).");
```
I like this appropriate indentation for wrapped lines

juneha1120 commented on src/main/java/Dodo.java:
```diff
@@ -2,111 +2,181 @@
+        }
+    }
+
+    public static void validTaskNumber(int i) throws DodoException {
```
Shouldn't this method name be in verbs?

potatodudedude committed:
Add ability to write to file when adding task

potatodudedude committed:
Add delete from storage when deleting tasks

potatodudedude committed:
Change method of updating storage
Add done status to storage

Before, the method of updating storage was to copy the old save file and
tweak it during delete operations

Now, I added a getStorageString() method to the Task classes so that
updating can be done by iterating through the arrayList of tasks and
calling their getStorageString().

Given the old method needed to iterate through every line in the save
file anyway, this should not affect performance much whilst simplifying
the updating process

potatodudedude committed:
Add handling for incorrect marking storage

potatodudedude committed:
Useless commit just to see Sourcetree branch

potatodudedude committed:
Tweak for style

potatodudedude committed:
Merge branch 'branch-level-7'

potatodudedude committed:
Useless commit to show branching

potatodudedude committed:
Update all classes to use LocalDateTime instead of a string to store
time

potatodudedude committed:
Made checks for expired tasks and invalid event timings

potatodudedude committed:
Extract printing list to its own method

potatodudedude committed:
Add due command which returns list of tasks ending on the input date

potatodudedude committed:
Merge branch 'branch-level-8'

potatodudedude committed:
Useless commit to see branching

potatodudedude committed:
Add non-static run command to run the main body instead of running in
main

Add TaskList class and update methods accordingly

potatodudedude committed:
Add storage and TimeStringUtility classes to separate methods and
extract common methods and constants when dealing with LocalDateTime
and LocalDate

potatodudedude committed:
Add UI class and move all printed lines apart from ones from errors to
it

potatodudedude committed:
Add Parse class, moved all appropriate methods there, edited error
messages to be outputted from UI class

potatodudedude committed:
Add Command abstract class and subclasses, add DodoCheck class

Before, all the checks for correct input format were in Parse class,
and parsing was handled in Dodo class itself.

For the sake of more OOP and clearer class roles, I moved all checks to
a new class DodoCheck.
Parse class is now only responsible for parsing and returning the
appropriate commands.
Command subclasses allow for cleaner code, makes parsing process more
readable without lumping everything in one method.

potatodudedude committed:
Merge branch 'branch-A-MoreOOP'

potatodudedude committed:
Useless commit to show branching

potatodudedude committed:
Add command, task, utility packages

potatodudedude committed:
Merge branch 'branch-A-Packages'

potatodudedude committed:
Merge branch 'add-gradle-support' of https://github.com/potatodudedude/ip

potatodudedude committed:
Add checkstyle support and gradle tweaks

potatodudedude committed:
Useless commit for branching

potatodudedude committed:
Add Junit Testing for Parse and TimeStringUtility classes

potatodudedude committed:
Merge branch 'branch-A-Junit'

potatodudedude committed:
Add jar file configs for dodo.jar

potatodudedude committed:
Merge branch 'branch-A-Jar'

potatodudedude committed:
Write JavaDoc for all non-trivial public methods in dodo package

potatodudedude committed:
Tweaked coding standard for TimeStringUtility class(excluding JavaDoc)

potatodudedude committed:
Write JavaDoc for all classes

potatodudedude committed:
Change code to match coding standard(minus JavaDoc)

potatodudedude committed:
Merge branch 'branch-A-CodingStandard'

potatodudedude committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/dodo/task/Deadline.java
#	src/main/java/dodo/task/Event.java
#	src/main/java/dodo/task/Todo.java
#	src/main/java/dodo/utilities/DodoCheck.java
#	src/main/java/dodo/utilities/DodoException.java
#	src/main/java/dodo/utilities/TimeStringUtility.java

potatodudedude committed:
Add find command

potatodudedude committed:
Merge branch 'branch-Level-9'

Fixed conflicts
Standardised the parsing segment for the add command to match the rest.
Removed a redundant check for valid parsing format.
Updated some print outputs.

MakiseKurisuX commented on src/main/java/dodo/Dodo.java:
```diff
@@ -0,0 +1,63 @@
+    private UI ui;
+
+    /**
+     * Constructor that initialises with specified storage poth.
```
Comment typo: path instead of poth.

MakiseKurisuX commented on src/main/java/dodo/Storage.java:
```diff
@@ -0,0 +1,138 @@
+    /**
+     * Checks if storage file and directory exists, if not, it is created.
+     */
+    public void existenceCheck() {
```
existence is a noun. Perhaps you could rename this as checkForStorage or something else?

MakiseKurisuX commented on src/main/java/dodo/Storage.java:
```diff
@@ -0,0 +1,138 @@
+     *
+     * @throws DodoException Throws when input is not "T" or "F"
+     */
+    private boolean stringToBoolean(String line) throws DodoException {
```
Perhaps you could rename this method to be more concise on what you are returning. Perhaps parseTaskStatus?

MakiseKurisuX commented on src/main/java/dodo/Storage.java:
```diff
@@ -0,0 +1,138 @@
+     * @throws FileNotFoundException
+     * @throws DodoException Throws when storage file format is not valid
+     */
+    public void readTo(TaskList tasks) throws FileNotFoundException, DodoException {
```
Ambiguous method name. Perhaps you could rename this to readTasksFromFile?

MakiseKurisuX commented on src/main/java/dodo/task/TaskList.java:
```diff
@@ -0,0 +1,80 @@
+ * All methods for manipulating tasks are here.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
```
I like that you avoided the common mistake and named this 'tasks' instead of 'task' to show it is a list of tasks.

MakiseKurisuX commented on src/main/java/dodo/UI.java:
```diff
@@ -0,0 +1,182 @@
+     */
+    public void updateFind(String line) {
+        System.out.println("Here are the tasks matching your description of: " + line);
+    }
```
In my opinion, the method name convention of updateSomething seems a bit ambiguous. Perhaps these can be renamed to something else?

Example. updateDelete, not sure if it is updating or deleting. It could be renamed as displayDeletedTask?

MakiseKurisuX commented on src/main/java/dodo/UI.java:
```diff
@@ -0,0 +1,182 @@
+            dodoheadCount = 0;
+            System.out.println("You seem to be struggling. Type 'help' if you need to see the command list.");
+        }
+    }
```
Not sure what is being checked here. Perhaps you could rename this as something else? checkUserStatus for example.

juneha1120 reviewed (commented):
Overall I like how your code is neat and clear

potatodudedude committed:
Useless commit to show branching

potatodudedude committed:
Useless commit to show branching

potatodudedude committed:
Merge branch 'branch-A-CheckStyle'

Checkstyle was already implemented a while back with gradle, so
this branch is just for show.

potatodudedude committed:
Add javafx tutorial template

potatodudedude committed:
Integrate GUI into existing commands

All commands now work with GUI except for exit command.
renamed UI methods to be more in line with coding format

potatodudedude committed:
Fix stored tasks not persisting

potatodudedude committed:
Add intro message to GUI

potatodudedude committed:
Integrate bye command into GUI

potatodudedude committed:
Useless commit to show branching

potatodudedude committed:
Merge branch 'branch-level-10'

potatodudedude committed:
Add assertions for commands that use type

Add, invalid and mark commands use an integer type for certain
functions.

These integers have a set range of values. Falling outside of these
ranges will cause issues.

Let's add assertions to check ensure they always fall within
their values.

These variables are not influenced directly by the user, so any invalid
input is indicative of a program bug, not user error.

So, using an assertion to handle it seems appropriate, as opposed to
using it to check something like command typos, where we want the user
to be able to correct their command instead of closing down the program.

potatodudedude committed:
AddCommand: Change variable name and extract addTask method
FindCommand: Fix find list not showing

AddCommand has some code in its execute method that was shared across
all Task cases.

This causes code duplication.

Let's pull out the AddCommand code that was common across cases.

This allows the execute method to be shorter and more readable

Let's also fix a small bug that causes the Find command to not show
the result list properly

potatodudedude committed:
AddCommand: Extract out task adding from execute method

AddCommand's execute method has all the code for determining time
validity and parsing taskDescriptions.

This makes execute command rather long and not readable.

Let's extract out everything under each case into a separate method.

This makes execute command shorter and more understandable, and allows
me to more easily find which segment is responsible for which task type.

potatodudedude committed:
Rename methods and variables to fit coding standard in command classes

Some command class variables and method names were quite vague/not
appropriate.

Let's change that so it fits the standard.

potatodudedude committed:
DodoCheck: Update method names to fit coding standard
DueCommand: Make due command return event if date falls on event's
entire time range, not just the end date

potatodudedude committed:
Change update method in Storage to have name more in line with coding
standard

potatodudedude committed:
Merge pull request #2 from potatodudedude/branch-A-Assertions

Add assertions for commands that use type

potatodudedude committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/dodo/command/InvalidCommand.java
#	src/main/java/dodo/command/MarkCommand.java

potatodudedude committed:
DueCommand: Rename contents variable to to fit coding standard
FindCommand: Rename contents variable to fit coding standard
Space out certain chunks in some longer methods for better readability
Update the name change of the updateTaskListFromStorage method

potatodudedude committed:
Merge pull request #3 from potatodudedude/branch-A-CodeQuality

Branch a code quality

potatodudedude committed:
Add reminder function

The user has to manually enter today's date to find tasks due today.

This might be a hassle since checking tasks due that day may be a
commonly repeated action.

Let's add an automatic reminder that tells the user tasks due today on
startup.

This allows the user to be more aware of their deadlines without
further input from them.

potatodudedude committed:
Useless commit to show branching

potatodudedude committed:
Merge branch 'branch-BCD-Extension'

potatodudedude committed:
Add temporary UI.png for website

potatodudedude committed:
Make send button anchored when resizing

Send button stays at its location when window is resized.

This leads to awkward looking UI as the button floats in the middle.

Let's make it so the button stays at the bottom right corner.

This allows it to be easy to find without being intrusive after resizes.

potatodudedude committed:
Set scroll pane to stretch with resizing
Set minimum width and height for resizing

Scroll pane does not resize with window.
Window can be shrunk as small as possible.

This makes resizing look awkward when scroll pane doesn't follow the
window.
Window shrinking too small makes the text impossible to see and the app
unusable.

Let's make the scroll pane stretch with the window.
Let's limit the minimum shrinking of the window.

This allows the scroll pane to look good when resizing.
This prevents the window size from causing visual issues.

potatodudedude committed:
Add in css support for GUI

GUI is rather bland.

This makes it look boring and may reduce user satisfaction.

Let's add css support for the GUI.

This allows us to more easily beautify the GUI in the future.

potatodudedude committed:
Tweak GUI textboxes

potatodudedude committed:
Make user and dodo text boxes different colour
Make dodo text colour change based on certain commands

potatodudedude committed:
Change GUI background, text font, icons to be more aesthetic

potatodudedude committed:
Add TextColourPair utility class and changed message colouring

Dodo messages only change colour based on command type.
This is due to colour information being passed through a private String
in each command class object.

Even if there is an error message, the colour is still based on the
command type. This is not very flexible.
Implementation also leads to a lot of redundant coding for commands
that don't need a colour change.

Let's use the UI class to return a TextColourPair class instead to pass
both a text String and a colour String.

This allows the text colour and contents to be centralised into one
class, and also lets the changing of individual messages colour under
more flexible circumstances.

potatodudedude committed:
Merge branch 'branch-A-BetterGui'

potatodudedude committed:
Useless commit to show branching

Personality already done in previous commits.

potatodudedude committed:
Merge branch 'branch-A-Personality'

potatodudedude committed:
Useless commit to show branching

Suggested error handling already implemented in earlier commits.

potatodudedude committed:
Merge branch 'branch-A-MoreErrorHandling'

potatodudedude committed:
Unabstract basic dodoheadCount operation in UI

potatodudedude committed:
Update Ui.png and write README

potatodudedude committed:
Update Ui.png and user guide

potatodudedude committed:
Change find command to be case insensitive

potatodudedude committed:
Update help command

potatodudedude committed:
Fix bug with find command not handling invalid inputs

potatodudedude committed:
Update README
